pkg/providers/equinix.go

```diff
@@ -0,0 +1,121 @@
+package providers
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/packethost/packngo" //requires env var PACKET_AUTH_TOKEN
+)
+
+const (
+	hostPrefix string = "cir"
+)
+
+type equinixProviderConfig struct {
+	ProjectID string   `json:"projectid"` //project id in equinix
+	Facility  []string `json:"facility"`  //datacenters to use
+	Plan      string   `json:"plan"`      //server size
+	OS        string   `json:"os"`        //OS to install
+}
+
+type equinixProvider struct {
+	config equinixProviderConfig
+	client *packngo.Client
+}
+
+func EquinixProviderFactory(providerInfo string, secretData map[string][]byte) (Provider, error) {
+	config := equinixProviderConfig{
+		ProjectID: "f2c6f0a6-4673-464b-878d-cce8dc8d0648",
+		Facility:  []string{"da11", "dc10", "dc13", "ny5", "sv15"},
+		Plan:      "c3.small.x86",
+		OS:        "rocky_8",
+	}
+
+	if configJson, ok := secretData["config"]; ok {
+		if err := json.Unmarshal(configJson, &config); err != nil {
+			return nil, err
+		}
+	}
+
+	client, err := packngo.NewClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &equinixProvider{
+		config: config,
+		client: client,
+	}, nil
+}
+
+func (p *equinixProvider) Acquire() (Resource, error) {
+	resource := Resource{}
+
+	uniqueId := strings.Replace(uuid.New().String(), "-", "", -1)
+	resourceName := fmt.Sprintf("%s-%s", hostPrefix, uniqueId)
```

Comment:
Instead of using a unique ID here we could base the  "resourceName" on the CIR somehow so that it consistent.
Then before creating the device, check if a device with the same name already exists. If it does we could either
use it, delete it or just fail.

This would ensure we only ever create a single device per CIR